src/parser/serializer.ts

```diff
@@ -42,30 +41,28 @@ export interface SerializeOptions {
 const regexp = /\x00/; // eslint-disable-line no-control-regex
 const ignoreKeys = new Set(['$db', '$ref', '$id', '$clusterTime']);
 
+function setUtf8String(buffer: Uint8Array, stringToSet: string, offset: number): number {
```

Comment:
Created this as a replacement for the `isArray ? utf8 : ascii` logic for keys. There is no difference between utf8 and ascii serialization so the condition was unnecessary. The logic is also the same for setting strings inside an existing buffer so I use it for the BSON values that are strings as well.